Server/Components/Recordings/recordings_main.cpp

```diff
@@ -92,7 +92,7 @@ class RecordingsComponent final : public IRecordingsComponent, public PlayerConn
 			}
 
 			// Write on foot recording data
-			if (data->file_.good() && data->type_ == PlayerRecordingType_OnFoot)
+			if (data->type_ == PlayerRecordingType_Driver && data->file_.good())
```

Comment:
Should be `data->type_ == PlayerRecordingType_OnFoot` for the on foot recording data.